executor/gp.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package poolmgr
+package executor
```

Comment:
Same comment as above.. we should keep a clear package boundary around the poolmgr, which we could do by moving it into its own package.